Add python 3.11 to CI

Indeed, we hit a problem like this (notice that this is only for CI, we don't know whether it would work in real life):
```python traceback
fractal_server/app/runner/_slurm/executor.py:520 FractalSlurmExecutor._completion
    outdata: (
        b'\x80\x05\x95\xb0\x05\x00\x00\x00\x00\x00\x00\x89\x8c\x17cloudpickle.cloudpickle\x94\x8c\x14_make_skeleton_class\x94\x93\x94(\x8c\x08builtins\x94\x8c\x04type\x94\x93\x94\x8c\x0eExceptionProxy\x94h\x03\x8c\x06obje'
        b'ct\x94\x93\x94\x85\x94}\x94\x8c\n'
        b'__module__\x94\x8c\x08__main__\x94s\x8c 7aeb4d0fe0624bae924337e16c9ec0b1\x94Nt\x94R\x94\x8c\x1ccloudpickle.cloudpickle_fast\x94\x8c\x0f_class_setsta'
        b'te\x94\x93\x94h\x0f}\x94(h\x0bh\x0c\x8c\x07__doc__\x94Xj\x01\x00\x00\n'
        b'    Proxy class to serialise exceptions\n'
        b'\n'
        b'    In general exceptions are not serialisable. This proxy class saves the\n'
        b'    serialisable content of an exception. On the receiving end, it can be used\n'
        b'    to reconstruct a TaskExecutionError.\n'
        b'\n'
        b'    Attributes:\n'
        b'        exc_type_name: Name of the exception type\n'
        b'        tb: TBD\n'
        b'        args: TBD\n'
        b'        kwargs: TBD\n'
        b'    \x94\x8c\x08__init__\x94h\x00\x8c\x0e_make_function\x94\x93\x94(h\x00\x8c\r'
        b'_builtin_type\x94\x93\x94\x8c\x08CodeType\x94\x85\x94R\x94(K\x03K\x00K\x00K\x05K\x02KOC\x1e|\x01j\x00|\x00_\x01|\x02|\x00_\x02|\x03|\x00_\x03|\x04|\x00_\x04d\x00S\x00\x94N\x85\x94(\x8c\x08__name__\x94\x8c\r'
        b'exc_type_name\x94\x8c\x02tb\x94\x8c\x04args\x94\x8c\x06kwargs\x94t\x94(\x8c\x04self\x94\x8c\x08exc_type\x94h"h#h$t\x94\x8cQ/usr/local/lib/python3.9/dist-packages/frac'
        b'tal_server/app/runner/_slurm/remote.py\x94h\x16K*C\x08\x00\x03\x08\x01\x06\x01\x06\x01\x94))t\x94R\x94}\x94(\x8c\x0b__package__\x94\x8c fractal_server.app.runner._slu'
        b'rm\x94h h\x0c\x8c\x08__file__\x94h)uNNNt\x94R\x94h\x10\x8c\x12_function_setstate\x94\x93\x94h2}\x94}\x94(h h\x16\x8c\x0c__qualname__\x94\x8c\x17ExceptionProxy.__init__\x94\x8c\x0f__'
        b"annotations__\x94}\x94(h'\x8c\t_operator\x94\x8c\x07getitem\x94\x93\x94\x8c\x06typing\x94\x8c\x04Type\x94\x93\x94h\x03\x8c\r"
        b'BaseException\x94\x93\x94\x86\x94R\x94h"h=h>\x8c\x04List\x94\x93\x94h\x03\x8c\x03str\x94\x93\x94\x86\x94R\x94u\x8c\x0e__kwdefaults__\x94N\x8c\x0c__defaults__\x94Nh\x0bh\x0ch\x14N\x8c\x0b__closure__\x94N\x8c\x17_'
        b'cloudpickle_submodules\x94]\x94\x8c\x0b__globals__\x94}\x94u\x86\x94\x86R0\x8c\r'
        b'__slotnames__\x94]\x94u}\x94\x86\x94\x86R0)\x81\x94}\x94(h!\x8c\tTypeError\x94h"\x8c8TypeError: code() takes at most 16 arguments (18 given)\n'
        b'\x94h#\x8c,code() takes at most 16 arguments (18 given)\x94\x85\x94h$}\x94ub\x86\x94.'
    ) (bytes) len=1467
FAILED                                                                                                                                                            [100%]
tests/test_backend_slurm.py:109 test_slurm_executor_submit - TypeError: code() argument 13 must be str, not int…



                from devtools import debug
                debug(outdata)
>               success, output = cloudpickle.loads(outdata)
E               TypeError: code() argument 13 must be str, not int
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -75,9 +75,9 @@ jobs:
         python-version: ["3.9", "3.10", "3.11"]
         db: ["sqlite", "postgres", "postgres-psycopg"]
         exclude:
-          - python_version: "3.11"
+          - python-version: "3.11"
             db: "sqlite"
-          - python_version: "3.11"
+          - python-version: "3.11"
             db: "postgres"
 
     services:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,7 +15,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.11"]
+        python-version: ["3.9", "3.10"]
         db: ["postgres"]
 
     services:
@@ -72,8 +72,13 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.11"]
+        python-version: ["3.9", "3.10", "3.11"]
         db: ["sqlite", "postgres", "postgres-psycopg"]
+        exclude:
+          - python_version: 3.11
+            db: sqlite
+          - python_version: 3.11
+            db: postgres
 
     services:
       postgres:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -75,10 +75,10 @@ jobs:
         python-version: ["3.9", "3.10", "3.11"]
         db: ["sqlite", "postgres", "postgres-psycopg"]
         exclude:
-          - python_version: 3.11
-            db: sqlite
-          - python_version: 3.11
-            db: postgres
+          - python_version: "3.11"
+            db: "sqlite"
+          - python_version: "3.11"
+            db: "postgres"
 
     services:
       postgres:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,7 +15,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.11"]
         db: ["postgres"]
 
     services:
@@ -72,7 +72,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.11"]
         db: ["sqlite", "postgres", "postgres-psycopg"]
 
     services:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@
     * Do not test V1 API with `DB_ENGINE="postgres-psycopg"` (\#1667).
     * Use new Fractal SLURM containers in CI (\#1663).
     * Adapt tests so that they always refer to the current Python version (the one running `pytest`), when needed; this means that we don't require the presence of any additional Python version in the development environment, apart from the current one (\#1633).
+    * Include Python3.11 in some tests (\#1669).
 
 # 2.3.3
 

--- a/tests/v1/04_api/test_admin_api.py
+++ b/tests/v1/04_api/test_admin_api.py
@@ -462,14 +462,10 @@ async def test_view_job(
         assert len(res.json()) == 0
 
         # get jobs by status
-        res = await client.get(
-            f"{PREFIX}/job/?status={JobStatusTypeV1.FAILED}"
-        )
+        res = await client.get(f"{PREFIX}/job/?status=failed")
         assert res.status_code == 200
         assert len(res.json()) == 0
-        res = await client.get(
-            f"{PREFIX}/job/?status={JobStatusTypeV1.SUBMITTED}"
-        )
+        res = await client.get(f"{PREFIX}/job/?status=submitted")
         assert res.status_code == 200
         assert len(res.json()) == 2
 

--- a/tests/v2/03_api/test_api_admin.py
+++ b/tests/v2/03_api/test_api_admin.py
@@ -168,14 +168,10 @@ async def test_view_job(
         assert len(res.json()) == 0
 
         # get jobs by status
-        res = await client.get(
-            f"{PREFIX}/job/?status={JobStatusTypeV2.FAILED}"
-        )
+        res = await client.get(f"{PREFIX}/job/?status=failed")
         assert res.status_code == 200
         assert len(res.json()) == 0
-        res = await client.get(
-            f"{PREFIX}/job/?status={JobStatusTypeV2.SUBMITTED}"
-        )
+        res = await client.get(f"{PREFIX}/job/?status=submitted")
         assert res.status_code == 200
         assert len(res.json()) == 2
 

